model.go

```diff
@@ -0,0 +1,150 @@
+// model.go
+
+package main
+
+import (
+	"database/sql"
+	// tom: errors is removed once functions are implemented
+	// "errors"
+)
+
+
+// tom: add backticks to json
+type product struct {
+	ID    int     `json:"id"`
+	Name  string  `json:"name"`
+	Price float64 `json:"price"`
+}
+
+// tom: these are added after tdd tests
+func (p *product) getProduct(db *sql.DB) error {
+	return db.QueryRow("SELECT name, price FROM products WHERE id=$1",
+		p.ID).Scan(&p.Name, &p.Price)
+}
+
+func (p *product) updateProduct(db *sql.DB) error {
+	_, err :=
+		db.Exec("UPDATE products SET name=$1, price=$2 WHERE id=$3",
+			p.Name, p.Price, p.ID)
+
+	return err
+}
+
+func (p *product) deleteProduct(db *sql.DB) error {
+	_, err := db.Exec("DELETE FROM products WHERE id=$1", p.ID)
+
+	return err
+}
+
+func (p *product) createProduct(db *sql.DB) error {
+	err := db.QueryRow(
+		"INSERT INTO products(name, price) VALUES($1, $2) RETURNING id",
+		p.Name, p.Price).Scan(&p.ID)
+
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func getProducts(db *sql.DB, start, count int) ([]product, error) {
```

Comment:
not required, maybe for now we can have a single api for fetching all products in the store